### DISCUSSION TITLE
How many domains should a student get in touch with?

### DISCUSSION THREAD

**Discussion by MakisH** [the-teachingRSE-project/RSE-Masters#39] [General]
Besides the RSE skills, a student already has their home domain from their bachelor's (see #1). Should they formally get in touch with a third domain as well?

Should a Chemistry graduate studying RSE also get a course on History?

**Comment by juliandehne** [comment]
I think RSE education and one Domain would be sufficient. I am not clear why the student should change the domain in the masters? Possibly, I missed the discussion here.

**Comment by jpthiele** [comment]
I think a student should be allowed to, if the number of courses permits, but I would not make it mandatory.
For example my computational engineering curricula (B.Sc. and M.Sc.) had  a 30 ECTS "Wahlpflicht" module for engineering with the constraint that 15 ECTS should be done in one of the fields, i.e. mechanical, electrical and civil eng.
I could envision something similar for the domain courses but it highly depends on the amount of ECTS we have available for it.

**Comment by jpthiele** [comment]
OTOH I find it important to do the collaborative project in a has-to-be-mixed team.

**Comment by CaptainSifff** [comment]
My opinion, was ,that if people want to go down to a specialized $DOMAIN-RSE, which probably requires continuing some of the $DOMAIN Master courses, then a single intro into another domain is sufficient. If they aim for a career in e.g. an RSE-unit( @juliandehne  you probably haven't seen this: https://github.com/DE-RSE/2023_paper-RSE-groups  ), 
where they might cater to multiple domains, they should probably aim for two "different" fields from their home domain.
How far away? If you have done mechanical engineering before, then e.g. electrical engineering and biology would be OK.

**Comment by knarrff** [comment]
You could see this a bit similar to "which CS topics should a CS master student have to take"? I am under the impression that while a Masters study is the start of when you specialize into something (in your thesis), you still stay quite general in your overall spread of topics. This should be similar here. One of the things a degree does is to give future employees a rough idea of what someone learned and should be expected to be able to do, at Masters level with a slight specialization, with emphasis on "slight".

What I would expect an RSE masters to know is, probably not exhaustive:
- a solid foundation in practical math (not as much as a CS, physics or even math students)
- a basic understanding of science concepts (that is: STEM or MINT in German): natural laws and scientific processes in those areas
- a basic understanding of qualitative research: typical workflows in the humanities
- some basic understanding of general issues like licensing, data protection laws and issues
- at least basic proficiency in all major programming languages and frameworks; at the moment this would at least include
    - python with numpy and matplotlib
    - R with ggplot2
    - julia
    - C, C++
Why this last list here? Because which is important also depends on the domain.

In the end, I should be able to hire an RSE master into any RSE group and get some "well rounded package". If I am more interested in a specific topic, I might look into someone with a specific thesis topic, but still get an "RSE generalist". When I am interested into an RSE **really deeply** into topic X, I am likely not going to pick an RSE masters, but will look for a Master in that domain and teach them missing RSE skills later: that's a lot easier to do than the other way around.

**Comment by juliandehne** [comment]
I agree with both perspectives. I think the question is what the study program is supposed to qualify for. University has the issue that it targets mainly science and other jobs are sort second plan. In that case, @CaptainSifff argument that domain science should be a valid profile holds. The well-rounded RSE generalist might be more like a specialization of a CS program. However, if it "only" contains the general skill set, a one-year master might be sufficient and definiately does not need a domain science at all. Maybe we can use CS-RSE and MNT-RSE as references for the two main ideas/profiles for the future discussion?

**Comment by juliandehne** [comment]
The "institutional constraints" kick-off workshop is going to deal with some of these aspects: https://terminplaner6.dfn.de/de/b/b87d7614820fb9540532386d855885ce-1253209.
